pkg/resources/jobs/starter.go

```diff
@@ -40,8 +39,20 @@ func NewStarterJob(k6 *v1alpha1.K6, hostname []string) *batchv1.Job {
 	if k6.Spec.Starter.AutomountServiceAccountToken != "" {
 		automountServiceAccountToken, _ = strconv.ParseBool(k6.Spec.Starter.AutomountServiceAccountToken)
 	}
-	command, istioEnabled := newIstioCommand(k6.Spec.Scuttle.Enabled, []string{"sh", "-c"})
-	env := newIstioEnvVar(k6.Spec.Scuttle, istioEnabled)
+
+	if k6.Spec.ServiceMesh.Istio.Enabled {
+		command = newIstioCommand(k6.Spec.ServiceMesh.Istio.Enabled, command)
+		image = "ghcr.io/grafana/operator:latest-starter-istio"
+	} else if k6.Spec.ServiceMesh.Linkerd.Enabled {
+		command = newLinkerdCommand(k6.Spec.ServiceMesh.Linkerd.Enabled, command)
+		image = "ghcr.io/grafana/operator:latest-starter-linkerd"
+	}
+
```

Comment:
This part can be simplified too once `BaseContainer` is added: similarly to runner.